test/service_tests.cpp

```diff
@@ -57,3 +61,21 @@ BOOST_AUTO_TEST_CASE( put_bucket) {
     free_client(client);
     BOOST_CHECK(error == NULL);
 }
+
+BOOST_AUTO_TEST_CASE( get_system_information ) {
+    ds3_client* client = get_client();
+    ds3_request* request = ds3_init_get_system_information();
+    ds3_get_system_information_response* response;
+
+    printf("-----Testing GET system_information-------\n");
+
+    ds3_error* error = ds3_get_system_information(client, request, &response);
+    BOOST_CHECK(error == NULL);
+
+    BOOST_CHECK(response->api_version != NULL);
+    BOOST_CHECK(response->serial_number != NULL);
```

Comment:
in the sample response in the self documentation those fields were empty, but a curl request against a live system does indeed show info in those fields so I'll add those asserts